Tools/GNUMake/comps/gnu.mak

```diff
@@ -105,6 +105,10 @@ ifeq ($(WARN_ALL),TRUE)
     warning_flags += -Wshadow
   endif
 
+  ifeq ($(gcc_major_version),7)
+    warning_flags += -Wno-array-bounds
```

Comment:
offline discussed: GCC 5-6 fine, GCC 7 false positives, GCC9+ fine

Reply:
I also tested GCC 8.4.  It works fine.  Note that it is an known issue that GCC 8.1-8.3 crashes when compiling AMReX.  So we need to avoid GCC 8.1-8.3.